pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1971,6 +1971,14 @@ func (ws *WorkflowStatus) TaskResultsInProgress() bool {
 	return false
 }
 
+func (ws *WorkflowStatus) IsTaskResultInCompleted(name string) bool {
+	value, found := ws.TaskResultsCompletionStatus[name]
```

Comment:
We should probably fix + update the comment for `ws.TaskResultsCompletionStatus` in `workflow_types.go`.

```
// TaskResultsCompletionStatus tracks task result completion status (mapped by pod name). Used to prevent premature archiving and garbage collection.
```

Maybe we could change it to the following?
```
// TaskResultsCompletionStatus tracks task result completion status (mapped by node ID). Used to prevent premature archiving, garbage collection, and workflow progression.
```

Reply:
Yes, may we can unified to nodeId.